src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,56 +1,128 @@
+  const [newMovieInfo, setNewMovieInfo] = useState(initialFormState);
+
+  const {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  } = newMovieInfo;
+
+  const handleChange = (formField: string | number, value: string | number) => {
+    setNewMovieInfo((pevMovieInfo) => ({
+      ...pevMovieInfo,
+      [formField]: value,
+    }));
+  };
+
+  const clearForm = () => {
+    setNewMovieInfo(initialFormState);
+  };
+
+  const trimTextField = (textField: string): string => textField.trim();
+
+  const isAllRequiredFieldsFilled = () => {
+    return (
+      trimTextField(title)
+      && trimTextField(imgUrl)
+      && trimTextField(imdbUrl)
+      && trimTextField(imdbId)
+    );
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onAdd(newMovieInfo);
+    setCount((prevCount) => prevCount + 1);
+    clearForm();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => handleSubmit(event)}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(newValue) => handleChange('title', newValue)}
         required
       />
 
       <TextField
         name="description"
         label="Description"
-        value=""
+        value={description}
+        onChange={(newValue) => handleChange('description', newValue)}
       />
 
       <TextField
         name="imgUrl"
         label="Image URL"
-        value=""
+        value={imgUrl}
+        onChange={(newValue) => handleChange('imgUrl', newValue)}
+        required
       />
 
       <TextField
         name="imdbUrl"
         label="Imdb URL"
-        value=""
+        value={imdbUrl}
+        onChange={(newValue) => handleChange('imdbUrl', newValue)}
+        required
       />
 
       <TextField
         name="imdbId"
         label="Imdb ID"
-        value=""
+        value={imdbId}
+        onChange={(newValue) => handleChange('imdbId', newValue)}
+        required
       />
 
       <div className="field is-grouped">
         <div className="control">
-          <button
-            type="submit"
-            data-cy="submit-button"
-            className="button is-link"
-          >
-            Add
-          </button>
+          {isAllRequiredFieldsFilled()
+            ? (
+              <button
+                type="submit"
+                data-cy="submit-button"
+                className="button is-link"
+              >
+                Add
+              </button>
+            ) : (
+              <button
+                type="submit"
+                data-cy="submit-button"
+                className="button is-link"
+                disabled
+              >
+                Add
+              </button>
+            )}
```

Comment:
we change only `disable` props here
```suggestion
              <button
                type="submit"
                data-cy="submit-button"
                className="button is-link"
                disabled={isAllRequiredFieldsFilled()}
              >
                Add
              </button>

```